src/test/javascript/spec/component/programming-exercise/category-issues-chart.component.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { CategoryIssuesChartComponent } from 'app/exercises/programming/manage/grading/charts/category-issues-chart.component';
+import { StaticCodeAnalysisCategory, StaticCodeAnalysisCategoryState } from 'app/entities/static-code-analysis-category.model';
+import { NgbTooltip } from '@ng-bootstrap/ng-bootstrap';
+import { MockDirective } from 'ng-mocks';
+
+describe('CategoryIssuesChartComponent', () => {
+    let fixture: ComponentFixture<CategoryIssuesChartComponent>;
+    let comp: CategoryIssuesChartComponent;
+    let category: StaticCodeAnalysisCategory;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [],
+            declarations: [CategoryIssuesChartComponent, MockDirective(NgbTooltip)],
+            providers: [],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CategoryIssuesChartComponent);
+                comp = fixture.componentInstance;
+            });
+    });
+
+    const initComponent = (categories: StaticCodeAnalysisCategory, maxGradedIssues: number, totalStudents: number, maxNumberOfIssues: number) => {
+        comp.issuesMap = configureIssuesMap();
+        comp.category = category;
+        comp.maxGradedIssues = maxGradedIssues;
+        comp.totalStudents = totalStudents;
+        comp.maxNumberOfIssues = maxNumberOfIssues;
+
+        comp.ngOnChanges();
+        tick();
+    };
+
+    const configureCategory = () => {
+        category = {
+            id: 0,
+            name: '',
+            state: StaticCodeAnalysisCategoryState.Graded,
+            penalty: 2,
+            maxPenalty: 4,
+        };
+        return category;
```

Comment:
[nit] I think either you assign the category here and use it in the test like this:
```ts
it('test', () => {
  configureCategory();
  testFunc();
});
```

Or you return the category without assigning it to the `category` variable and do it in the test, like:
```ts
const configureCategory = () => {
  return {
    id: 0,
    name: '',
    state: StaticCodeAnalysisCategoryState.Graded,
    penalty: 2,
    maxPenalty: 4,
  };
}

it('test', () => {
  let category = configureCategory();
  testFunc();
});
```

Because it looks like you have different usages from test to test.